app/validators/presence.js

```diff
@@ -21,14 +22,24 @@ export default Base.extend({
   },
 
   validate(value, options) {
-    if (options.presence === true && isEmpty(value)) {
+    if (options.presence === true && !this._isPresent(value)) {
       return this.createErrorMessage('blank', value, options);
     }
 
-    if(options.presence === false && !isEmpty(value)) {
+    if(options.presence === false && this._isPresent(value)) {
       return this.createErrorMessage('present', value, options);
     }
 
     return true;
+  },
+
+  /**
+   * Handle presence of ember proxy based instances
+   */
+  _isPresent(value) {
+    if(Ember.ObjectProxy.detectInstance(value) || Ember.ArrayProxy.detectInstance(value)) {
```

Comment:
i would recommend using the raw JS `instanceof` because this is what `detectInstance` now becomes
https://github.com/emberjs/ember.js/blob/d38cadd99aa1fbeea5b603b4eff2b94fb3fe74ec/packages/ember-runtime/lib/system/core_object.js#L811-L813

We plan to likely remove detectInstance in 3.0